src/api/axiosRequest.js

```diff
@@ -0,0 +1,78 @@
+import axios from "axios";
+
+export async function AxiosRequest(url, method, headers, params) {
+  return params
+    ? axios({
+        //work for POST
+        //if paramtrs comes then take call to the param ..otherwise go downside with blank object
+        url: url,
+        method: method,
+        headers: headers, //id provides extra data also req,n response or provide metadata  .
+        data: params,
+        timeout: 1000,
+      })
+    : axios({
+        //works for a GET
+        url: url,
+        method: method,
+        headers: headers,
+        params: {},
+        timeout: 1000,
+      });
+}
+
+const GetApiDetails = () => {
+  const headers = {
```

Comment:
why don't you create headers at global level and use them, and if you want to add new headers , use spread operator.

const defaultHeaders = {
  'Content-Type': 'application/json',
};

  return AxiosRequest("http://localhost:3001/details", "GET", headers: {
      ...defaultHeaders }, {});